src/models/inbound/dfspConsentRequests.model.ts

```diff
@@ -0,0 +1,281 @@
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Sridhar Voruganti - sridhar.voruganti@modusbox.com
+ --------------
+ ******/
+
+import { PubSub } from '~/shared/pub-sub'
+import { PersistentModel } from '~/models/persistent.model'
+import { StateMachineConfig } from 'javascript-state-machine'
+import { ThirdpartyRequests, Errors } from '@mojaloop/sdk-standard-components';
+import {
+  v1_1 as fspiopAPI,
+  thirdparty as tpAPI
+} from '@mojaloop/api-snippets'
+import inspect from '~/shared/inspect'
+import { HTTPResponseError } from '~/shared/http-response-error'
+import {
+  DFSPConsentRequestsData,
+  DFSPConsentRequestsStateMachine,
+  DFSPConsentRequestsModelConfig
+} from '~/models/inbound/dfspConsentRequests.interface'
+import { DFSPBackendRequests } from '~/shared/dfsp-backend-requests'
+
+// DFSPConsentRequestsModel is the passive inbound handler for inbound
+// POST /consentRequests requests and no response is generated from `model.run()`
+export class DFSPConsentRequestsModel
+  extends PersistentModel<DFSPConsentRequestsStateMachine, DFSPConsentRequestsData> {
+  protected config: DFSPConsentRequestsModelConfig
+
+  constructor (
+    data: DFSPConsentRequestsData,
+    config: DFSPConsentRequestsModelConfig
+  ) {
+    const spec: StateMachineConfig = {
+      init: 'start',
+      transitions: [
+        { name: 'validateRequest', from: 'start', to: 'RequestIsValid' },
+        { name: 'storeReqAndSendOTP', from: 'RequestIsValid', to: 'success' }
+      ],
+      methods: {
+        // specific transitions handlers methods
+        onValidateRequest: () => this.onValidateRequest(),
+        onStoreReqAndSendOTP: () => this.onStoreReqAndSendOTP()
+      }
+    }
+    super(data, config, spec)
+    this.config = { ...config }
+  }
+
+  // getters
+  get pubSub (): PubSub {
+    return this.config.pubSub
+  }
+
+  get dfspBackendRequests (): DFSPBackendRequests {
+    return this.config.dfspBackendRequests
+  }
+
+  get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  async onValidateRequest (): Promise<void> {
+    const { request, toParticipantId } = this.data
+
+    try {
+      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+
+      if (!response) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR
+        )
+      }
+
+      if (!response.isValid) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_NO_SUPPORTED_SCOPE_ACTIONS
```

Comment:
Can you really infer this error from `response.isValid`? That seems like a bit of a logical leap to me

Reply:
As discussed backend response contains validation details

Reply:
Yep, all good.